File: 271-encode-and-decode-strings/271-encode-and-decode-strings.ts
/**
 * Encodes a list of strings to a single string.
 */
function encode(strs: string[]): string {
	return JSON.stringify(strs)
};

/**
 * Decodes a single string to a list of strings.
 */
function decode(s: string): string[] {
	return JSON.parse(s)
};

/**
 * Your functions will be called as such:
 * decode(encode(strs));
 */

File: 271.encode-and-decode-strings.py
#
# @lc app=leetcode id=271 lang=python3
#
# [271] Encode and Decode Strings
#

# @lc code=start
class Codec:
    def encode(self, strs: List[str]) -> str:
        """Encodes a list of strings to a single string.
        """
        return ''.join(f"{len(s)}#{s}" for s in strs)

    def decode(self, s: str) -> List[str]:
        """Decodes a single string to a list of strings.
        """
        res = []
        i = 0
        while i < len(s):
            j = i
            while s[j] != '#':
                j += 1
            length = int(s[i:j])
            res.append(s[j+1:j+1+length])
            i = j + 1 + length
        return res

# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(strs))
# @lc code=end

